sklearn/decomposition/_incremental_pca.py

```diff
@@ -234,15 +234,19 @@ def partial_fit(self, X, y=None, check_input=True):
         self : object
             Returns the instance itself.
         """
+        first_pass = not hasattr(self, "components_")
         if check_input:
             if sparse.issparse(X):
                 raise TypeError(
                     "IncrementalPCA.partial_fit does not support "
                     "sparse input. Either convert data to dense "
                     "or use IncrementalPCA.fit to do so in batches.")
-            X = check_array(X, copy=self.copy, dtype=[np.float64, np.float32])
+            X = self._validate_data(
+                X, accept_sparse=['csr', 'csc', 'lil'],
```

Comment:
Why do we need to pass `accept_sparse` here?

Reply:
Good catch. I copied and pasted from the fit method without properly thinking about it.